src/bot/bot_message.py

```diff
@@ -0,0 +1,47 @@
+"""Copyright by CookieDecide, Darkuuu
+Licensed under MIT License
+"""
+
+import discord
+from discord.ext import commands
+import logging
+import util.embed
+
+
+class BotMessage(commands.Cog):
+    """Collection of all Message commands and corresponding utility functions.
+
+    Attributes:
+        bot: Reference to discord bot.
+    """
+
+    def __init__(self, bot):
+        """Initializes the instance.
+
+        Args:
+            bot: Discord bot.
+        """
+        self.bot = bot
+
+    @commands.command(name="purge", help="Deletes the last x messages.")
+    async def purge(self, ctx: commands.Context, count: int):
+        """Deletes the last x messages.
+
+        Args:
+            ctx: Context of command invocation.
+            count: Number of messages to delete.
+        """
+        logging.info(f"Received purge request from user {ctx.author}")
+
+        deleted = 0
+        to_delete = count
+        while to_delete > 100:
+            await ctx.channel.purge(limit=100)
+            deleted = deleted + 100
```

Comment:
Would following scenario make the calculations wrong?:

User requests to purge 120 (>100) message, the channel only contains 50 messages (<100). Code goes into while loop since only request amount matters. Here, only 50 messages can be deleteded but deleted and to_delete adds/subtracts 100, therefore final amount is wrong.

I guess checking the amount of messages which got deleted in the loop and then handle accordingly (like done below) could fix that.

Reply:
Should be fixed now.